buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy

```diff
@@ -138,9 +138,10 @@ class NodeInfo {
         String esJavaOpts = config.jvmArgs.isEmpty() ? collectedSystemProperties : collectedSystemProperties + " " + config.jvmArgs
         env.put('ES_JAVA_OPTS', esJavaOpts)
         for (Map.Entry<String, String> property : System.properties.entrySet()) {
-            if (property.getKey().startsWith('es.')) {
+            assert property.key.startsWith('es.') == false : "BUILD MISCONFIGURED: Use -Dtests.es.* to pass system properties to ES. -Des.* is not supported explicitly"
```

Comment:
I added it to help to catch CI misconfigurations so that they get corrected. Happy to remove though since the only misconfigured one seems to have been fixed already.